ledger/catchpointtracker.go

```diff
@@ -266,70 +268,94 @@ func (ct *catchpointTracker) committedUpTo(rnd basics.Round) (retRound, lookback
 	return rnd, basics.Round(0)
 }
 
-func (ct *catchpointTracker) produceCommittingTask(committedRound basics.Round, dbRound basics.Round, dcr *deferredCommitRange) *deferredCommitRange {
-	var hasMultipleIntermediateCatchpoint, hasIntermediateCatchpoint bool
-
-	newBase := dcr.oldBase + basics.Round(dcr.offset)
-
-	// catchpoint for round X = oldBase + offset + lookback
-	// is taken at X - CatchpointLookback
-	// check if X - CatchpointLookback is within oldBase...oldBase+offset
-	if ct.catchpointInterval > 0 {
-		nextCatchpointRound := ((uint64(dcr.oldBase+dcr.lookback) + ct.catchpointInterval) / ct.catchpointInterval) * ct.catchpointInterval
-		if nextCatchpointRound > dcr.catchpointLookback {
-			snapshotRound := nextCatchpointRound - dcr.catchpointLookback
-			if snapshotRound < uint64(dcr.oldBase)+dcr.offset {
-				newBase = basics.Round(snapshotRound)
-				mostRecentCatchpointRound := (uint64(committedRound) / ct.catchpointInterval) * ct.catchpointInterval
-				if mostRecentCatchpointRound > dcr.catchpointLookback {
-					mostRecentSnapshotRound := mostRecentCatchpointRound - dcr.catchpointLookback
-					if mostRecentSnapshotRound > snapshotRound {
-						hasMultipleIntermediateCatchpoint = true
-						// skip if there is more than one catchpoint in queue
-						newBase = basics.Round(mostRecentSnapshotRound)
-					}
-					hasIntermediateCatchpoint = true
-				}
+// Calculate whether we have intermediate first stage catchpoint rounds and the
+// new offset.
+func calculateFirstStageRounds(oldBase basics.Round, offset uint64, accountDataResourceSeparationRound basics.Round, catchpointInterval uint64, catchpointLookback uint64) (bool /*hasIntermediateFirstStageRound*/, bool /*hasMultipleIntermediateFirstStageRounds*/, uint64 /*offset*/) {
```

Comment:
why comment these return variables?

Reply:
The comments are to document what each return value is.

Reply:
and Go has named variables for exact same reason. Please use named vars here and in another place I pointed out before.